app/models/doi.rb

```diff
@@ -1273,30 +1273,30 @@ def downloads_over_time
   end
 
   def reference_ids
-    reference_events.pluck(:target_doi).compact.uniq.map(&:downcase)
+    reference_events.pluck(:target_doi).compact.map(&:downcase).uniq
   end
 
   def reference_count
-    reference_events.pluck(:target_doi).uniq.length
+    reference_events.pluck(:target_doi).compact.map(&:downcase).uniq.length
   end
 
   def indexed_references
     Doi.query(nil, ids: reference_ids, page: { number: 1, size: 25 }).results
   end
 
   def citation_ids
-    citation_events.pluck(:source_doi).compact.uniq.map(&:downcase)
+    citation_events.pluck(:source_doi).compact.map(&:downcase).uniq
   end
 
   # remove duplicate citing source dois
   def citation_count
-    citation_events.pluck(:source_doi).uniq.length
+    citation_events.pluck(:source_doi).compact.map(&:downcase).uniq.length
   end
 
   # remove duplicate citing source dois,
   # then show distribution by year
   def citations_over_time
-    citation_events.pluck(:occurred_at, :source_doi).uniq { |v| v[1] }.
+    citation_events.pluck(:occurred_at, :source_doi).map { |v| [v[0], v[1].downcase] }.sort_by { |v| v[0] }.uniq { |v| v[1] }.
```

Comment:
@wendelfabianchinsamy I just added back the `pluck` because I think the `map` would have selected the entire Event record and may not have been performant for DOIs with many events. All good?